### PR TITLE
Missing config params on SFT

### PR DESCRIPTION
**Pull request by tcapelle**
Hi,
Small PR to add the missing warmup and the total number of steps so the training happens correctly.
I am also adding info on the GPU requirements ( 80GB Gpus ). <- this is on the main readme =P

image

The [link to the experiment](https://wandb.ai/capecape/zephyr/workspace)

### CODE REVIEW

File: recipes/zephyr-7b-beta/sft/config_lora.yaml [hunk 1]
```diff
@@ -37,8 +37,9 @@ log_level: info
 logging_steps: 5  
 logging_strategy: steps
 lr_scheduler_type: cosine
+warmup_ratio: 0.1
 max_seq_length: 2048
-max_steps: -1
+max_steps: 272
```
Comment (alvarobartt) [inline recipes/zephyr-7b-beta/sft/config_lora.yaml:42]: I think `max_steps=-1` because `num_train_epochs` is used instead

Comment (tcapelle) [reply recipes/zephyr-7b-beta/sft/config_lora.yaml:42]: Yeah, but the ConstantLengthDataset doesn't know how many steps it will run, so the scheduler can't setup the warmup cycle correctly

Comment (tcapelle) [reply recipes/zephyr-7b-beta/sft/config_lora.yaml:42]: I amy try to fix this in trl

**Review by lewtun — COMMENTED** [review]
Good catch with the logging steps @tcapelle ! There's an open PR to fix this in TRL here (https://github.com/huggingface/trl/pull/979), so I suggest we keep the YAML configs of the repo unchanged for now

File: recipes/zephyr-7b-beta/README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ As described in the Zephyr [technical report](https://huggingface.co/papers/2310
 See below for commands to train these models using either DeepSpeed ZeRO-3 or LoRA.
 
 ## Full training examples
-
+You will require 8 GPUs (80GB of VRAM) to train the full model.
```
Comment (lewtun) [inline recipes/zephyr-7b-beta/README.md:12]: Happy to keep this line in the PR if you don't mind reverting the config changes :)

Comment (tcapelle) [reply recipes/zephyr-7b-beta/README.md:12]: haha, my bad cause this is specified on the main readme file =)

Comment (lewtun) [reply recipes/zephyr-7b-beta/README.md:12]: thanks for iterating!

**Review by lewtun — APPROVED** [review]
